test/suites/integration/bsp/bsp-thresholds.test.ts

```diff
@@ -0,0 +1,175 @@
+import assert, { strictEqual } from "node:assert";
+import {
+  addBsp,
+  bspDownKey,
+  bspDownSeed,
+  bspThreeKey,
+  bspThreeSeed,
+  bspTwoKey,
+  bspTwoSeed,
+  describeBspNet,
+  type EnrichedBspApi,
+  ShConsts
+} from "../../../util";
+
+describeBspNet(
+  "BSPNet: BSP Volunteering Thresholds",
+  { initialised: false },
+  ({ before, it, createUserApi, beforeEach }) => {
+    let api: EnrichedBspApi;
+
+    before(async () => {
+      api = await createUserApi();
+    });
+
+    beforeEach(async () => {
+      await api.sealBlock(api.tx.sudo.sudo(api.tx.fileSystem.setGlobalParameters(1, 1)));
+    });
+
+    it("Can set params with setGlobalParams", async () => {
+      const { extSuccess } = await api.sealBlock(
+        api.tx.sudo.sudo(api.tx.fileSystem.setGlobalParameters(87, 200))
+      );
+
+      strictEqual(extSuccess, true, "Extrinsic should be successful");
+
+      strictEqual(
+        (await api.query.fileSystem.blockRangeToMaximumThreshold()).toNumber(),
+        200,
+        "Threshold should have changed"
+      );
+      strictEqual(
+        (await api.query.fileSystem.replicationTarget()).toNumber(),
+        87,
+        "Replication Target should have changed"
+      );
+    });
+
+    it("Shouldn't be able to setGlobalParams without sudo", async () => {
+      const { extSuccess, events } = await api.sealBlock(
+        api.tx.fileSystem.setGlobalParameters(13, 37)
+      );
+
+      strictEqual(extSuccess, false, "Extrinsic should be unsuccessful");
+      const { data } = api.assert.eventPresent("system", "ExtrinsicFailed", events);
+      const error = data[0].toString();
+      strictEqual(error, "BadOrigin", "Extrinsic should fail with BadOrigin");
+
+      strictEqual(
+        (await api.query.fileSystem.blockRangeToMaximumThreshold()).toNumber(),
+        1,
+        "Threshold should not have changed"
+      );
+      strictEqual(
+        (await api.query.fileSystem.replicationTarget()).toNumber(),
+        1,
+        "Replication Target should not have changed"
+      );
+    });
+
+    it("Reputation increased on successful storage", { skip: "Not Implemented" }, async () => {
+      const repBefore = (await api.query.providers.backupStorageProviders(ShConsts.DUMMY_BSP_ID))
+        .unwrap()
+        .reputationWeight.toBigInt();
+      await api.file.newStorageRequest("res/cloud.jpg", "test/cloud.jpg", "bucket-1");
+      await api.wait.bspVolunteer();
+      await api.wait.bspStored();
+
+      const repAfter = (await api.query.providers.backupStorageProviders(ShConsts.DUMMY_BSP_ID))
+        .unwrap()
+        .reputationWeight.toBigInt();
+
+      assert(
+        repAfter > repBefore,
+        "Reputation should increase after successful storage request fufilled"
+      );
+      console.log(`Reputation increased from ${repBefore} to ${repAfter}`);
+    });
+
+    it("zero reputation can still volunteer and be accepted", async () => {
+      // Create a new BSP and onboard with no reputation
```

Comment:
This actually shows something is wrong. A BSP shouldn't have 0 reputation. They start with 1. Good that you raised this.

Reply:
Replaced 0 rep with min (1), and changed default bsp to 5